main.tf

```
@@ -2,24 +2,113 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "~> 3.27"
+      version = "~> 3.0"
     }
   }
-
-  required_version = ">= 0.14.9"
 }
 
 provider "aws" {
-  profile = "default"
-  region  = "us-east-1"
+  region = "us-east-1"
+}
+
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+ 
+ tags = {
+    Name = "DevOps_VPC"
+  }
+}
+
+locals {
+    public_cidr = ["10.0.0.0/24", "10.0.1.0/24"]
+    private_cidr = ["10.0.2.0/24", "10.0.3.0/24"]
+}
+
+resource "aws_subnet" "public" {
+    count = 2
```

This code will only work if we have strictly 2 subnets. However, the task says "at least 2"
The following trick will allow you to change number of subnets without having to change much code
```suggestion
  count = length(local.public_cidr)
```

More info here: https://www.terraform.io/language/functions/length

Please make the change here and in all other places where count is used.